samples/drivers/w1/scanner/prj.conf

```diff
@@ -0,0 +1,5 @@
+CONFIG_I2C=y
+CONFIG_LOG=y
+CONFIG_SERIAL=y
+CONFIG_W1=y
+CONFIG_W1_DS248X=y
```

Comment:
remove CONFIG_W1_DS248X, will get set by `default ` in Kconfig change.

Reply:
Not sure why, but this doesn't work on my machine.

Reply:
That's probably because you are missing defining `DT_COMPAT_MAXIM_DS248X` in the Kconfig.  If you look at the `drivers/counter/Kconfig.esp32` as an example, you'll see that it has `DT_COMPAT_ESP32_TIMER := espressif,esp32-timer`